src/main/archetype/it.tests/src/main/java/it/tests/HtmlUnitClient.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2020 Adobe Systems Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ${package}.it.tests;
+
+import com.adobe.cq.testing.client.CQClient;
+import com.gargoylesoftware.htmlunit.DefaultCssErrorHandler;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.WebClientOptions;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.sling.testing.clients.ClientException;
+import org.apache.sling.testing.clients.SlingClientConfig;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Node;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.fail;
+
+/**
+ * AEM client that maintains a WebClient instance from HttpUnit framework
+ */
+public class HtmlUnitClient extends CQClient {
```

Comment:
This class looks really helpful. Wouldn't it make sense to add that to https://github.com/adobe/aem-testing-clients instead. It doesn't seem that this isn't really project specific.... That way one could update the class easily and extend functionality without being forced to change code in the project SCM.

Reply:
I am not totally sure if this type of functionality really belongs on that page. I could imagine that a generic HtmlUnitClient helps in the TestingClients, but the functionality in here is quite specific to this usecase.